spark-schema-registry/src/main/scala/com/hortonworks/spark/registry/util/SchemaRegistryUtil.scala

```diff
@@ -0,0 +1,284 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hortonworks.spark.registry.util
+
+import java.io.ByteArrayInputStream
+import java.util.concurrent.atomic.AtomicInteger
+
+import scala.collection.mutable
+import com.hortonworks.registries.schemaregistry.{SchemaMetadata, SchemaVersionInfo, SchemaVersionKey}
+import com.hortonworks.registries.schemaregistry.client.SchemaRegistryClient
+import com.hortonworks.registries.schemaregistry.client.SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL
+import com.hortonworks.registries.schemaregistry.serdes.avro.{AvroSnapshotDeserializer, AvroSnapshotSerializer}
+import collection.JavaConverters._
+
+import com.hortonworks.spark.registry.sparkavro.{AvroDeserializer, AvroSerializer, SchemaConverters}
+import org.apache.avro.Schema
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.sql.api.java.UDF1
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.{BinaryType, DataType, StructType}
+
+/**
+ * Spark schemas and deserialization based on
+ * schema registry (https://github.com/hortonworks/registry) schemas.
+ *
+ * @param session the spark session
+ * @param config the schema registry config
+ */
+class SchemaRegistryUtil(session: SparkSession, config: Map[String, Object]) {
+  private val counter: AtomicInteger = new AtomicInteger()
+  private val srClient = new SchemaRegistryClient(config.asJava)
```

Comment:
Does `SchemaRegistry` provide some mini-server for some integration test?
Otherwise, could you add some UnitTest by using `mockito` for `SchemaRegistryClient`?

Reply:
Theres no mini server. Let me check if theres a possibility to launch an instance or else will add unit tests with mockito.